Adds C4B-Ansible initial documentation (#853)
chocolatey/docs: input/en-us/c4b-environments/ansible/client-setup.md

---
Order: 42

> xref: c4b-ansible-client-setup

Reviewer: Probably out of scope, but possibly for a part 2... client setup with Ansible? If I've got a fleet of servers to on-board and I've used Ansible to setup the server then chances are my go to will be Ansible for configuring the rest of the fleet too.

Title: Client Setup
Description: How to setup a client machine to use Chocolatey for Business Ansible Environment
---

## Summary

Once you have your environment deployed, you'll need to get clients talking to it.  
To do that, you'll need to do the following on the clients:

1. Ensure the client machine can access the C4B environment on ports 24020 and 8443 (by default).
1. [Install the SSL/TLS certificate](xref:c4b-azure-client-setup#ssl-certificate), if it was self-signed, so clients can access HTTPS components.
1. Install Chocolatey components and configure the client for Chocolatey for Business (C4B) deployments.

## Client Setup Script

> To install the Chocolatey components and on-board clients, you should run the `ClientSetup.ps1` script provided with your Chocolatey for Business Ansible Environment.

Reviewer: Where do I find the script? On the Ansible host, or in Nexus, or?

You will need the following values ready when running this script:

* `FQDN`: The fully qualified domain name used to access your environment.

> * `ccm_client_salt`: This is the client-side salt additive. More information about this can be found in the [C4B Config Settings](xref:ccm-client#config-settings) docs. The value will have been provided during the deployment of the C4B environment.

Reviewer 2: On the docs site, we have a Chocolatey For Business environment drop down, which houses the docs for our Azure marketplace image "environment", as well as the Quick Start guide "environment". I think we need to be consistent and consider this our "Ansible environment". The environment being "the complete collection of services to get the most out of your Chocolatey For Business installation".

> * `ccm_client_salt`: This is the client-side salt additive. More information about this can be found in the [C4B Config Settings](xref:ccm-client#config-settings) docs. The value will have been provided during the deployment of the C4B environment.

Reviewer 3: So that would be:

> The value will have been provided during the deployment of the Chocolatey for Business Ansible environment.

Or

> The value will have been provided during the Ansible deployment of the Chocolatey for Business environment.

I prefer the last one?

> * `ccm_client_salt`: This is the client-side salt additive. More information about this can be found in the [C4B Config Settings](xref:ccm-client#config-settings) docs. The value will have been provided during the deployment of the C4B environment.

Reviewer 3: Something I meant to say here. We are consistently using Chocolatey for Business (note the non-capitalized 'for').  So we should use it throughout. I am happy whichever we do it, but we just need to be consistent.

> * `ccm_service_salt`: This is the service salt additive. More information about this can be found in the [C4B Config Settings](xref:ccm-client#config-settings) docs. The value will have been provided during the deployment of the C4B environment.

Reviewer 2: Same comment from above here.

> * `ccm_service_salt`: This is the service salt additive. More information about this can be found in the [C4B Config Settings](xref:ccm-client#config-settings) docs. The value will have been provided during the deployment of the C4B environment.

Reviewer 3: Same options as above.

Same preference as above. :smile:

* `chocouser_password`: The password for the `chocouser` account which is used by the client to access your environment's Nexus component. The value will have been provided during the deployment of the C4B environment.

The values provided during the deployment were available in the `CCM.html` file provided in the `credentials` directory.

> :choco-info: **NOTE**
>
> You can set default values for the parameters and remove the Mandatory flag if you prefer to run the script without being prompted for input.

> When you're ready, run the following script on the client from an elevated (Run as Administrator) PowerShell console:

Author: @pauby Another thing for brief thought?

> When you're ready, run the following script on the client from an elevated (Run as Administrator) PowerShell console:

Reviewer 3: I would disagree you can elevate a shell, terminal or console. They are all tools. A shell is something that runs that you interact with. You can't elevate _it_. You can elevate _inside of it_. The same with terminal and console. A Terminal and console are something you interact with and is the tool you use to do that. You can't elevate them. 

You can, however, elevate a session within a shell running in a terminal or console. But those are, in my mind, strictly correct. People use these words interchangeably. I am trying to eliminate confusion and ensure we are consistent.

If it's something that the reader would immediately understand, then I don't think it really matters **as long as we are consistent**.

---
Order: 42
xref: c4b-ansible-client-setup
Title: Client Setup

> Description: How to setup a client machine to use Chocolatey for Business Ansible Environment

Reviewer 3: ```suggestion
Description: How to Setup a Client Machine to Use the Chocolatey for Business Ansible Environment
```

---

## Summary

> Once you have your environment deployed, you'll need to get clients talking to it.  

Reviewer 3: Should we say 'Ansible environment' here? I only ask to be consistent, but I'm fine either.

Author: To me, "ansible environment" doesn't make sense in this context (I'm not meaning The Product, here, and an ansible environment to me says AWX or a runner). I'll make sure I'm consistent with "environment" when not referring to the full titled product itself, "Chocolatey for Business Ansible Environment".

Reviewer 3: > 'Once you have your environment deployed'.

If you don't mean the Ansible environment, what environment are you referring to here?

To do that, you'll need to do the following on the clients:

> 1. Ensure the client machine can access the Chocolatey Central Management instance on port 24020, and the Sonatype Nexus instance on 8443 (by default).

Reviewer 3: ```suggestion
1. Ensure the client computer can access the Chocolatey Central Management instance on port 24020, and the Sonatype Nexus instance on 8443 (by default).
```

1. [Install Chocolatey components and configure the client for Chocolatey for Business](xref:c4b-ansible-client-setup#client-setup-script) (C4B) deployments and management.

## Client Setup Script

To install the Chocolatey components and on-board clients, you should run the `ClientSetup.ps1` script provided with your Chocolatey for Business Ansible Environment. By default, this script is stored in the newly created `choco-install` repository on the Sonatype Nexus instance.

You will need the following values ready when running this script:

* `FQDN`: The fully qualified domain name used to access your environment.
* `ccm_client_salt`: This is the client-side salt additive. More information about this can be found in the [C4B Config Settings](xref:ccm-client#config-settings) docs. The value will have been provided during the deployment of the C4B environment.
* `ccm_service_salt`: This is the service salt additive. More information about this can be found in the [C4B Config Settings](xref:ccm-client#config-settings) docs. The value will have been provided during the deployment of the C4B environment.

> * `chocouser_password`: The password for the `chocouser` account which is used by the client to access your environments' Sonatype Nexus service. The value will have been provided during the deployment of the C4B environment.

Reviewer 3: ```suggestion
* `chocouser_password`: The password for the `chocouser` account which is used by the client to access your environments' Sonatype Nexus service. The value will have been provided during the deployment of the Chocolatey for Business environment.
```

Reviewer 3: See above comments as this may affect what we put here.
